src/modes/mobile/IndicatorCorrelation.svelte

```diff
@@ -22,7 +32,34 @@
    */
   export let fetcher;
 
-  $: otherSensors = sensorList.filter((d) => d.key !== sensor.key);
+  /**
+   * @param {SensorEntry} sensorEntry
+   * @returns {Promise<CorrelationMetric>}
+   */
+  function buildMetrics(sensorEntry) {
+    const contextData = fetcher.fetch1Sensor1RegionNDates(sensor, region, date.windowTimeFrame);
+    const compareData = fetcher.fetch1Sensor1RegionNDates(new SensorParam(sensorEntry), region, date.windowTimeFrame);
+    return Promise.all([contextData, compareData]).then((p) => {
+      const metrics = generateCorrelationMetrics(p[0], p[1]);
+      return metrics;
+    });
+  }
+
+  /**
+   * Starting from a list of sensors, mix in all of the other columns needed for the table.
+   */
+  function buildTableData() {
+    return sensorList
+      .filter((d) => d.key !== sensor.key)
+      .map((sensor) => {
+        return {
+          ...sensor,
+          metrics: buildMetrics(sensor),
+        };
+      });
+  }
+
+  $: otherSensors = buildTableData();
```

Comment:
@sgratzl I don't have a full mental model of how Svelte handle reactive updates.  This renders fine on initial load, but then doesn't update based on the user's suggestion.

Reply:
the code in the function won't be tracked. So you have to pass all the tracked/reactive variables, in this case `sensor, region, date`

Reply:
Ah, I see.  It's only reactive with respect to the declarations that have changed. Makes sense.  Thanks.